R/pmcmc.R

```diff
@@ -119,20 +119,43 @@ pmcmc <- function(pars, filter, n_steps, save_state = FALSE,
     }
   }
 
-  pars <- set_colnames(list_to_matrix(history_pars$get()),
-                       names(curr_pars))
+  pars_matrix <- set_colnames(list_to_matrix(history_pars$get()),
+                              names(curr_pars))
   probabilities <- set_colnames(list_to_matrix(history_probabilities$get()),
                        c("log_prior", "log_likelihood", "log_posterior"))
 
-  state <- trajectories <- NULL
+  predict <- state <- trajectories <- NULL
   if (save_state) {
     state <- t(list_to_matrix(history_state$get()))
+
+    ## Extract the transformation function from the pars object as
+    ## it's all we want; this is a bit of a hack but it works for now
+    ## at least.
+    transform <- pars[[".__enclos_env__"]]$private$transform
```

Comment:
Haven't seen this before, just double-checking it's a safe way to get this? You can't alter the environment in some way that breaks it?

Reply:
Presumably this isn't possible within `predict_info()`?

Reply:
This relies on an implementation detail in R6, rather than an R detail, and is safe here insofar as R6 will not remove it (https://github.com/r-lib/R6/issues/122, and I think that is depended on fairly widely) and we control the parameters class.

I think that we'd be better off working out how to handle method-like fields in the transform object, or exposing the private field as a readonly public one.

Reply:
> Presumably this isn't possible within predict_info()?

`predict_info` only sees elements of the particle filter, but we need the transform function, which comes from the `pmcmc_parameters` object